src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -91,13 +100,75 @@ class Results extends React.Component<Props, State> {
     const currentQuery = eventView.getEventsAPIPayload(location);
     const prevQuery = prevState.eventView.getEventsAPIPayload(prevProps.location);
     if (!isAPIPayloadSimilar(currentQuery, prevQuery)) {
-      this.fetchTotalCount();
+      this.canLoadEvents();
     }
   }
 
-  async fetchTotalCount() {
+  canLoadEvents = async () => {
+    const {api, location, organization} = this.props;
+    const {eventView} = this.state;
+    let needConfirmation = false;
+    let confirmedQuery = true;
+    const currentQuery = eventView.getEventsAPIPayload(location);
+
+    const duration =
+      currentQuery.statsPeriod && currentQuery.statsPeriod.endsWith('d')
+        ? parseInt(currentQuery.statsPeriod.slice(0, -1), 10)
+        : (new Date(currentQuery.end) - new Date(currentQuery.start)) /
+          (24 * 60 * 60 * 1000);
+
+    if (duration >= 30) {
+      const {results} = await fetchProjects(api, organization.slug, {allProjects: true});
+      let projectLength = currentQuery.project.length;
+
+      if (projectLength === 0) {
+        // My Projects
+        projectLength = results.filter(project => project.isMember).length;
+      } else if (projectLength === 1 && currentQuery.project[0] === '-1') {
+        // All Projects
+        projectLength = results.length;
+      }
+
+      if (projectLength >= 10) {
+        needConfirmation = true;
+        confirmedQuery = false;
+      }
+    }
+    this.setState({needConfirmation, confirmedQuery});
+    this.fetchTotalCount(confirmedQuery);
+
+    // Any time confirmed Query is enabled, need to flip it off after the fetch starts
+    await doTimeout(0);
+    this.setState({confirmedQuery: false});
```

Comment:
Could you toggle the state before `fetchProjects` so that after you know the project length you can settle the state without an additional wait?

Reply:
Hmm not sure if I misunderstood, wouldn't toggling the state pre `fetchProjects` result in querying without confirmation?